src/formatDate.js

```diff
@@ -50,7 +50,39 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = fromFormat[3];
+  const newSeparator = toFormat[3];
+  const oldDate = date.split(oldSeparator);
+  const dateInfo = {};
+  const newDate = [];
+
+  for (let i = 0; i < oldDate.length; i++) {
+    if (fromFormat[i] === 'YY') {
+      dateInfo['YYYY'] = extendYearsFormat(oldDate[i]);
+    } else if (fromFormat[i] === 'YYYY') {
+      dateInfo['YY'] = shortYearsFormat(oldDate[i]);
+    }
+
+    dateInfo[fromFormat[i]] = oldDate[i];
+  }
```

Comment:
It seems like you could iterate using `for of` or `for each` loops)

A rule of thumb: if you don't need any manipulation with indices (terminating earlier, getting the previous/next element [i + 1]/[i - 1] etc.), use either `for of` or `for each` loops or something like that)

Reply:
Oops, I overlooked the last statement😅

Reply:
Agree. For of will be better 👌

Reply:
If ypu need to keep track of the current index, you could try `forEach` it accepts index as the third argument)